float64 start_x
float64 start_y
float64 end_x
float64 end_y
---
float32[] rx
float32[] ry